vyper/parser/events.py

```diff
@@ -108,24 +109,9 @@ def pack_args_by_32(holder, maxlen, arg, typ, context, placeholder,
         copier = make_byte_array_copier(dest_placeholder, source_lll, pos=pos)
         holder.append(copier)
         # Add zero padding.
-        new_maxlen = ceil32(source_lll.typ.maxlen)
-
-        holder.append([
-            'with', '_ceil32_end', ['ceil32', ['mload', dest_placeholder]], [
-                'seq', ['with', '_bytearray_loc', dest_placeholder, [
-                    'seq', ['repeat', zero_pad_i, ['mload', '_bytearray_loc'], new_maxlen, [
-                        'seq',
-                        # stay within allocated bounds
-                        ['if', ['ge', ['mload', zero_pad_i], '_ceil32_end'], 'break'],
-                        [
-                            'mstore8',
-                            ['add', ['add', '_bytearray_loc', 32], ['mload', zero_pad_i]],
-                            0,
-                        ],
-                    ]],
-                ]],
-            ]
-        ])
+        holder.append(
+            zero_pad(dest_placeholder, maxlen, zero_pad_i=zero_pad_i)
+        )
```

Comment:
@jacqueswww I do not think this refactoring is semantics-preserving.

In order to be semantics-preserving, the following two conditions need to be satisfied, but I have a counter-example that the first condition is not always met.

1. `maxlen` is equal to `new_maxlen`.
2. The runtime value of `_ceil32_end` is equal to `new_maxlen`.

Reply:
@charles-cooper I forgot to mention you for this. What do you think about this?

Reply:
I'm not sure. I thought `maxlen` will get `ceil32` applied to it in `zero_pad`.